SummerSchool/app/src/main/java/nl/rug/www/rugsummerschools/controller/MainActivity.java

```diff
@@ -0,0 +1,386 @@
+        }
+
+        @Override
+        public int getCount() {
+            return FRAGMENTS_SIZE;
+        }
+
+        @Override
+        public int getItemPosition(@NonNull Object object) {
+            if (object instanceof ForumLoginFragment || object instanceof ForumThreadListFragment) {
+                return POSITION_NONE;
+            }
+            return super.getItemPosition(object);
+        }
+    };
+
+    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
+            = new BottomNavigationView.OnNavigationItemSelectedListener() {
+
+        @Override
+        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
+            switch (item.getItemId()) {
+                case R.id.navigation_announcement:
+                    mViewPager.setCurrentItem(0);
+                    return true;
+                case R.id.navigation_general:
+                    mViewPager.setCurrentItem(1);
+                    return true;
+                case R.id.navigation_lecturer:
+                    mViewPager.setCurrentItem(2);
+                    return true;
+                case R.id.navigation_time_table:
+                    mViewPager.setCurrentItem(3);
+                    return true;
+                case R.id.navigation_forum:
+                    mViewPager.setCurrentItem(4);
+                    return true;
+            }
+            return false;
+        }
+    };
+
+    private ViewPager.SimpleOnPageChangeListener mSimpleOnPageChangeListener = new ViewPager.SimpleOnPageChangeListener() {
+        @Override
+        public void onPageSelected(int position) {
+            switch (position) {
+                case 0 :
+                    mNavigation.setSelectedItemId(R.id.navigation_announcement);
+                    break;
+                case 1 :
+                    mNavigation.setSelectedItemId(R.id.navigation_general);
+                    break;
+                case 2 :
+                    mNavigation.setSelectedItemId(R.id.navigation_lecturer);
+                    break;
+                case 3 :
+                    mNavigation.setSelectedItemId(R.id.navigation_time_table);
+                    break;
+                case 4 :
+                    mNavigation.setSelectedItemId(R.id.navigation_forum);
+                    break;
+            }
+            invalidateOptionsMenu();
+        }
+    };
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        Toolbar toolbar = findViewById(R.id.my_toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setTitle(null);
+
+        mNavigation = (BottomNavigationView) findViewById(R.id.navigation);
+        mNavigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
+        disableShiftingNavigationMode();
+
+        // initialize facebook login
```

Comment:
this method is a bit too long. 
it would make sense to extract methods from the //initialize fb login and //initialize goog login sections

Reply:
Make sense. Will refactor it.